python/hail/expr/aggregators/aggregators.py

```diff
@@ -49,8 +49,8 @@ def __init__(self):
                       ret_type=hail_type,
                       constructor_args=sequenceof(expr_any),
                       init_op_args=nullable(sequenceof(expr_any)),
-                      f=nullable(func_spec(1, expr_any)))
-    def __call__(self, name, aggregable, ret_type, constructor_args=(), init_op_args=None, f=None):
+                      additional_seq_op_args=nullable(oneof(sequenceof(expr_any), func_spec(1, expr_any))))
+    def __call__(self, name, aggregable, ret_type, constructor_args=(), init_op_args=None, additional_seq_op_args=None):
```

Comment:
OK, this code is a bit of a mess because the current aggregator design isn't quite right for multiple, but I think it can be improved.  First, `f` is terrible name and it's meaning is unclear.  I think we should have:

```
  def __call__(self, name, aggregable, ret_type, constructor_args=(), init_op_args=None, seq_op_args=()):
    ...
```

Then the seq_op and agg_sig can be constructed with:

```
        # no longer takes seq_op_args
        def agg_sig():
            return AggSignature(name,
                                list(map(get_type, constructor_args)),
                                None if init_op_args is None else list(map(get_type, init_op_args)),
                                list(map(get_type, seq_op_args)))

            def make_seq_op(agg):
                uid = Env.get_uid()
                ref = construct_variable(uid, get_type(agg), agg._indices)
                applied_seq_ops = [x(ref) if callable(x) else x for x in seq_op_args]
                return construct_expr(Let(uid, get_ir(agg), SeqOp(I32(0), [get_ir(x) for x in callable_seq_op_args], agg_sig())), None)
            seq_op = aggregable._transformations(aggregable, make_seq_op)
            signature = agg_sig()
```

Then we don't need to special case Count or test if `f` is `None`.  Any seq_op_arg can be an Expression or a callable that is expecting to take the aggregation value.  Count and the other aggregators that pass `f` will need to be modified slightly.

This is a bit gnarly so let's discuss in person if you have questions.